### PR TITLE
feat(rds): allow `DatabaseClusterFromSnapshot` to set `copyTagsToSnapshot` property

### LINKED ISSUE #19884: (aws-rds): DatabaseClusterFromSnapshot should set copyTagsToSnapshot
### Describe the bug

Today, the `copyTagsToSnapshot` property exists on `DatabaseClusterProps` not `DatabaseClusterBaseProps`: https://github.com/aws/aws-cdk/blob/789e8d2aaa0aefb6d17e4ebc0d56c17e9999add0/packages/%40aws-cdk/aws-rds/lib/cluster.ts#L497-L511

It's also not set in the `DatabaseClusterFromSnapshot` construct, so if you switch a `DatabaseCluster` to `DatabaseClusterFromSnapshot`, you'll get this diff:

```
[~] AWS::RDS::DBCluster CdkPipelinesPrototype-dev-us-east-1/Database/Database/cdk-pipeline-example-dev-Cluster DatabasecdkpipelineexampledevClusterC6927963 replace
 ├─ [-] CopyTagsToSnapshot
 │   └─ true
<trimmed>
```

This is especially noticeable because the value defaults to `true` in the L2 construct, which is different from the CloudFormation default (`false`): https://github.com/aws/aws-cdk/blob/789e8d2aaa0aefb6d17e4ebc0d56c17e9999add0/packages/@aws-cdk/aws-rds/lib/cluster.ts#L505-L510

Unlike other values that are explicitly unset when using `DatabaseClusterFromSnapshot` (e.g., `KmsKeyId`, `MasterUsername`, `MasterUserPassword` - see <https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-kmskeyid>), the [docs for `CopyTagsToSnapshot`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-copytagstosnapshot) don't specify that you shouldn't set the value when restoring from a snapshot.[](url)

### Expected Behavior

I expected to be able to switch from a `DatabaseCluster` to a `DatabaseClusterFromSnapshot` while retaining the same default behaviors.

### Current Behavior

Currently, the `CopyTagsToSnapshot` value shows as being unset (which will change to the CloudFormation default of `false`) when changing from a `DatabaseCluster` to `DatabaseClusterFromSnapshot`.

### Reproduction Steps

https://github.com/blimmer/cdk-bug-reports/compare/rds-copy-tags-to-snapshot?expand=1

In this simple stack:

```ts
import { Stack, StackProps } from 'aws-cdk-lib';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { AuroraPostgresEngineVersion, DatabaseCluster, DatabaseClusterEngine, DatabaseClusterFromSnapshot } from 'aws-cdk-lib/aws-rds';
import { Construct } from 'constructs';

export class CdkBugReportsStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const vpc = new Vpc(this, 'VPC')
    new DatabaseCluster(this, 'DatabaseCluster', {
      engine: DatabaseClusterEngine.auroraPostgres({ version: AuroraPostgresEngineVersion.VER_13_4 }),
      instanceProps: {
        vpc
      }
    })

    new DatabaseClusterFromSnapshot(this, 'DatabaseClusterFromSnapshot', {
      snapshotIdentifier: 'arn:aws:rds:us-east-1:12345678910:cluster-snapshot:rds:example-2022-04-12-09-26',
      engine: DatabaseClusterEngine.auroraPostgres({ version: AuroraPostgresEngineVersion.VER_13_4 }),
      instanceProps: {
        vpc
      }
    })
  }
}
```

The resource with the id `DatabaseCluster` has `CopyTagsToSnapshot` set to `true` (the default in the L2 construct):

```json
"DatabaseCluster68FC2945": {
  "Type": "AWS::RDS::DBCluster",
  "Properties": {
    "Engine": "aurora-postgresql",
    "CopyTagsToSnapshot": true,
```

However, the one restored from the snapshot does not have this value set:

```json
"DatabaseClusterFromSnapshotB2BFF6A0": {
  "Type": "AWS::RDS::DBCluster",
  "Properties": {
    "Engine": "aurora-postgresql",
	// CopyTagsToSnapshot is missing
```

### Possible Solution

I think `copyTagsToSnapshot` should be moved up to `DatabaseClusterBaseProps` and should default to `true` for `DatabaseClusterFromSnapshot` for consistency.

### Additional Information/Context

You can work around this temporarily by setting the value on the L1 construct, e.g.:

```ts
const db = new DatabaseClusterFromSnapshot(this, 'DatabaseClusterFromSnapshot', {
  snapshotIdentifier: 'arn:aws:rds:us-east-1:12345678910:cluster-snapshot:rds:example-2022-04-12-09-26',
  engine: DatabaseClusterEngine.auroraPostgres({ version: AuroraPostgresEngineVersion.VER_13_4 }),
  instanceProps: {
    vpc
  }
});
(db.node.defaultChild as CfnDBCluster).copyTagsToSnapshot = true;
```

### CDK CLI Version

2.20.0

### Framework Version

_No response_

### Node.js Version

16.14.0

### OS

macOS

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by AnuragMohapatra**
This change will now allow users to set `copyTagsToSnapshot` attribute for the `DatabaseClusterFromSnapshot`.

This will now be consistent with the way the database instance works.

_Integration test may not be possible since this requires a valid snapshot to pre-exist before building the CDK stack to generate the CDK local snapshot._

_It might be possible to add a snapshot first by creating a new DB Cluster taking a manual snapshot then using that in `DatabaseClusterFromSnapshot` but it will become a hassle for anyone else in future to maintain or update since they will always need to proceed in the same manner to get the exact snapshot._ 

_An integration test for `copyTagsToSnapshot` already exists for `DatabaseCluster` in `integ.cluster.js` which is passing successfully on executing the tests._

Closes #19884

----

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/lib/cluster.ts [hunk 1]
```diff
@@ -265,6 +265,12 @@ interface DatabaseClusterBaseProps {
   * @default - if storageEncrypted is true then the default master key, no key otherwise
   */
   readonly storageEncryptionKey?: kms.IKey;
+  /**
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/cluster.ts:268]: Let's add an empty line before this line.

File: packages/@aws-cdk/aws-rds/lib/cluster.ts [hunk 2]
```diff
@@ -265,6 +265,12 @@ interface DatabaseClusterBaseProps {
...
+  * Whether to copy tags to the snapshot when a snapshot is created.
+  *
+  * @default: true
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/cluster.ts:271]:
```suggestion
  * @default true
```

File: packages/@aws-cdk/aws-rds/lib/cluster.ts [hunk 3]
```diff
@@ -425,6 +431,8 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
       // Encryption
       kmsKeyId: props.storageEncryptionKey?.keyArn,
       storageEncrypted: props.storageEncryptionKey ? true : props.storageEncrypted,
+      //Tag
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/cluster.ts:434]:
```suggestion
      // Tags
```

File: packages/@aws-cdk/aws-rds/README.md [hunk 4]
```diff
@@ -63,6 +63,37 @@ new rds.DatabaseClusterFromSnapshot(this, 'Database', {
...
+    instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+    vpcSubnets: {
+      subnetType: ec2.SubnetType.PRIVATE,
+    },
+    vpc,
+  },
+  copyTagsToSnapshot: false
+});
+```
+
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/README.md:96]: I don't think this small change is worth all this text. Let's revert this change (I'll add a label to the PR to not require a change to a ReadMe file in this PR).

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the contribution @AnuragMohapatra! A few small comments.

File: packages/@aws-cdk/aws-rds/lib/cluster.ts [hunk 5]
```diff
@@ -265,6 +265,12 @@ interface DatabaseClusterBaseProps {
   * @default - if storageEncrypted is true then the default master key, no key otherwise
   */
   readonly storageEncryptionKey?: kms.IKey;
+  /**
+  * Whether to copy tags to the snapshot when a snapshot is created.
+  *
+  * @default: true
+  */
+  readonly copyTagsToSnapshot?: boolean;
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/cluster.ts:273]: This looks misaligned:

```suggestion
  /**
   * Whether to copy tags to the snapshot when a snapshot is created.
   *
   * @default: true
   */
  readonly copyTagsToSnapshot?: boolean;

File: packages/@aws-cdk/aws-rds/lib/cluster.ts (hunk 1 above)
Comment (AnuragMohapatra) [reply packages/@aws-cdk/aws-rds/lib/cluster.ts:269]: Added the empty line, also aligned the misaligned KMS comments

File: packages/@aws-cdk/aws-rds/lib/cluster.ts (hunk 2 above)
Comment (AnuragMohapatra) [reply packages/@aws-cdk/aws-rds/lib/cluster.ts]: Updated the comment as per suggestion.

File: packages/@aws-cdk/aws-rds/README.md (hunk 4 above)
Comment (AnuragMohapatra) [reply packages/@aws-cdk/aws-rds/README.md]: Removed the readme details.

File: packages/@aws-cdk/aws-rds/lib/cluster.ts (hunk 5 above)
Comment (AnuragMohapatra) [reply packages/@aws-cdk/aws-rds/lib/cluster.ts]: Updated the comment as per suggestion.

**Comment by AnuragMohapatra** [comment]
Thanks @skinny85 for reviewing, I have updated the changes as per suggestions.

**Review by skinny85 — APPROVED** [review]
Thanks for the contribution @AnuragMohapatra!

**Pull request merged**